src/lib/draw/Button.svelte

```diff
@@ -0,0 +1,57 @@
+<script lang="ts">
+  import type { Mode } from "./types";
+
+  export let mode: Mode;
+  export let thisMode: Mode;
+  export let label: string;
+  export let icon: string;
+  export let changeMode: (m: Mode) => void;
+</script>
+
+<button
+  type="button"
+  on:click={() => changeMode(thisMode)}
+  disabled={mode == thisMode}><img src={icon} alt={label} /> {label}</button
+>
+
+<style>
+  button {
+    background-color: #4caf50;
+    border: solid 2px #4caf50;
+    color: white;
+    padding: 15px 32px;
+    text-align: center;
+    text-decoration: none;
+    display: inline-block;
+    font-size: 16px;
+    margin: 4px 2px;
+    cursor: pointer;
+    width: 100%;
+  }
+
+  button:hover:enabled {
+    background-color: white;
+    color: black;
+    transition-duration: 0.4s;
+    border: solid 2px black;
+  }
+
+  button:disabled {
+    opacity: 0.6;
+    cursor: auto;
+  }
+
+  /* The SVG icons are defined with black. By default, make them white, using
+   * https://codepen.io/sosuke/pen/Pjoqqp to calculate the filter. */
+  img {
+    filter: invert(100%) sepia(0%) saturate(0%) hue-rotate(328deg)
+      brightness(100%) contrast(101%);
+    vertical-align: middle;
```

Comment:
![Screenshot from 2023-05-15 15-09-55](https://github.com/acteng/atip/assets/1664407/b4ebf4df-26cf-4551-a2aa-5cd3d4533d7e)
This isn't what I wanted, but couldn't figure out how to CSS this properly